File: Isomorphic-strings/Isomorphic-strings.py
#leetcode Isomorphic trings - easy

class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        pairs = {}
        values = set()
        for char_s, char_t in zip(s, t):
            if char_s not in pairs:
                if char_t in values:
                    return False
                pairs[char_s] = char_t
                values.add(char_t)
            else:
                if char_t != pairs.get(char_s):
                    return False
        return True

File: Isomorphic-strings/Isomorphic-strings.js
//leetcode - Isomorphic Strings - easy

var isIsomorphic = function(s, t) {
    const pairs = new Map();
    const tValues = new Set();
    for (let i = 0; i < s.length; i++) {
        if (!pairs.has(s[i])) {
            if (tValues.has(t[i])) {
                return false
            };
            pairs.set(s[i], t[i]);
            tValues.add(t[i])
        } else {
            if (t[i] !== pairs.get(s[i])) {
                return false
            }
        }
    };
    return true
};